frontend/src/app/main-table/cost-table/cost-table.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from "@angular/core";
+import { Subscription } from "rxjs";
+
+import { NamespaceService } from "src/app/services/namespace.service";
+import { ExponentialBackoff } from "src/app/utils/polling";
+import { KubecostService, AggregateCostResponse } from 'src/app/services/kubecost.service';
+
+@Component({
+  selector: "app-cost-table",
+  templateUrl: "./cost-table.component.html",
+  styleUrls: ["./cost-table.component.scss", "../main-table.component.scss"]
+})
+export class CostTableComponent implements OnInit {
+
+  // Logic data
+  aggregatedCost: AggregateCostResponse = null;
+  currNamespace = "";
+
+  subscriptions = new Subscription();
+  poller: ExponentialBackoff;
+
+  constructor(
+    private namespaceService: NamespaceService,
+    private kubecostService: KubecostService,
+  ) { }
+
+  ngOnInit() {
+    // Create the exponential backoff poller
+    this.poller = new ExponentialBackoff({ interval: 2000, retries: 3 });
+
+    // Keep track of the selected namespace
+    const namespaceSub = this.namespaceService
+      .getSelectedNamespace()
+      .subscribe(this.onNamespaceChange.bind(this));
+
+    this.subscriptions.add(namespaceSub);
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  onNamespaceChange(namespace: string) {
+    this.currNamespace = namespace;
+    this.poller.reset();
+    this.getAggregatedCost();
+  }
+
+  formatCost(value: number): string {
+    return '$' + (value > 0 ? Math.max(value, 0.01) : 0).toFixed(2)
+  }
+
+  getAggregatedCost() {
+    this.kubecostService.getAggregateCost(this.currNamespace).subscribe(
```

Comment:
Don't make this component responsible for fetching data. Instead, follow the pattern used in the main table component and existing children.